modules/account-lib/src/coin/sol/transaction.ts

```diff
@@ -41,21 +44,145 @@ export class Transaction extends BaseTransaction {
   }
 
   /** @inheritdoc */
-  canSign(key: BaseKey): boolean {
-    throw new NotImplementedError('canSign not implemented');
+  canSign(): boolean {
+    return true;
   }
 
-  sign(keyPair: KeyPair | KeyPair[]): void {
-    throw new NotImplementedError('sign not implemented');
+  /**
+   * Signs transaction.
+   *
+   * @param {KeyPair} keyPair Signer keys.
+   */
+  async sign(keyPair: KeyPair[] | KeyPair): Promise<void> {
+    if (!this._solTransaction || !this._solTransaction.recentBlockhash) {
+      throw new SigningError('Nonce is required before signing');
+    }
+    if (!this._solTransaction || !this._solTransaction.feePayer) {
+      throw new SigningError('feePayer is required before signing');
+    }
+    const keyPairs = keyPair instanceof Array ? keyPair : [keyPair];
+    const signers: Signer[] = [];
+    for (const kp of keyPairs) {
+      const keys = kp.getKeys(true);
+      if (!keys.prv) {
+        throw new SigningError('Missing private key');
+      }
+      signers.push({ publicKey: new PublicKey(keys.pub), secretKey: keys.prv as Uint8Array });
+    }
+    try {
+      this._solTransaction.partialSign(...signers);
+    } catch (e) {
+      throw e;
+    }
   }
 
   /** @inheritdoc */
   toBroadcastFormat(): string {
-    throw new NotImplementedError('toBroadcastFormat not implemented');
+    if (!this._solTransaction) {
+      throw new ParseTransactionError('Empty transaction');
+    }
+    // The signatures can have null signatures (which means they are required but yet unsigned)
+    // In order to be able to serializer the txs, we have to change the requireAllSignatures based
+    // on if the TX is fully signed or not
+    const requireAllSignatures = requiresAllSignatures(this._solTransaction.signatures);
+    try {
+      return this._solTransaction.serialize({ requireAllSignatures }).toString('base64');
+    } catch (e) {
+      throw e;
+    }
+  }
+
+  /**
+   * Sets this transaction payload
+   *
+   * @param rawTransaction
+   */
+  fromRawTransaction(rawTransaction: string): void {
+    try {
+      isValidRawTransaction(rawTransaction);
+      this._solTransaction = SolTransaction.from(Buffer.from(rawTransaction, 'base64'));
+
+      if (this._solTransaction.signature && this._solTransaction.signature !== null) {
+        this._id = base58.encode(this._solTransaction.signature);
+      }
+      const transactionType = getTransactionType(this._solTransaction);
+      switch (transactionType) {
+        case TransactionType.WalletInitialization:
+          this.setTransactionType(TransactionType.WalletInitialization);
+          break;
+        case TransactionType.Send:
+          this.setTransactionType(TransactionType.Send);
+          break;
```

Comment:
No default cases for error or notify ? Why if we have a new type of tx?

Reply:
capturing this comment in https://bitgoinc.atlassian.net/browse/STLX-10065
let's work on it as a follow up - there's other improvements we can make